lib/iris/util.py

```diff
@@ -406,16 +406,16 @@ def between(lh, rh, lh_inclusive=True, rh_inclusive=True):
         return lambda c: lh < c < rh
 
 
-def reverse(array, axes):
+def reverse(cube_or_array, axes):
     """
-    Reverse the array along the given axes.
+    Reverse the cube or array along the given axes.
 
     Args:
 
-    * array
-        The array to reverse
-    * axes
-        A single value or array of values of axes to reverse
+    * cube_or_array: :class:`iris.cube.Cube` or :class:`numpy.ndarray`
+        The cube or array to reverse
+    * axes: int or sequence of ints
```

Comment:
It's great that `reverse` can be used on cubes or NumPy arrays. This does, however, introduce a usage issue in that typically if I pass a cube to functionality that operates on a dimension I can specify the dimension of interest as a coordinate name as well as an axis. 

It would be good if you could pass one or more coordinates or axes here, and have the function look up the axis from the specified coordinate (I'm pretty sure there are examples of doing such elsewhere in the codebase). Of course, with this you'll need to add more checking, for example if you pass a NumPy array and named dimensions... You'll also need to add testing for this functionality.